tests/conftest.py

```diff
@@ -225,11 +225,35 @@ def action_example() -> Action:
     return action_factory()
 
 
+@pytest.fixture
+def action_example_with_inner_match() -> Action:
+    return action_factory(whiteboard_tag="inner-match")
+
+
+@pytest.fixture
+def action_example_without_brackets_required() -> Action:
+    return action_factory(whiteboard_tag="bracket-less", brackets_required=False)
+
+
 @pytest.fixture
 def actions_example(action_example) -> Actions:
     return Actions.parse_obj([action_example])
 
 
+@pytest.fixture
+def actions_example_with_inner_match(
+    action_example_with_inner_match,
+) -> Actions:
+    return Actions.parse_obj([action_example_with_inner_match])
+
+
+@pytest.fixture
+def actions_example_without_brackets_required(
+    action_example_without_brackets_required,
+) -> Actions:
+    return Actions.parse_obj([action_example_without_brackets_required])
+
+
```

Comment:
If these fixtures are only used in one or two tests, it may be better to define them at the module level, rather than clutter the root `conftest.py`.

Similarly, it looks like `action_example_with_inner_match` is only used to build `actions_example_with_inner_match`. Maybe we can just do

```python
@pytest.fixture
def actions_example_with_inner_match() -> Actions:
    action = action_factory(whiteboard_tag="inner-match")
    return Actions.parse_obj([action])
```

repeating that pattern for `action_example_without_brackets_required` / `actions_example_without_brackets_required`.

Reply:
I'm also noticing that these factories would be more useful if we had fixtures that simply returned the factory function, e.g. 

```
@pytest.fixture
def action_factory() -> Action:
    return factories.action_factory
```

Then we could use them directly in tests.

Reply:
```
@pytest.fixture
def action_factory() -> Action:
    return factories.action_factory
```

Initially there was some overloading issues when trying this out; but I modified the import practices to allow for this naming convention (and usage in the tests).

As for the repeating of tests, the newest commit is aimed to address that concern as well.